hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/RemoteEventsListener.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) 2020 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.ui;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.hawkbit.ui.common.event.EntityModifiedEventPayload;
+import org.eclipse.hawkbit.ui.common.event.EntityModifiedEventPayloadIdentifier;
+import org.eclipse.hawkbit.ui.common.event.EventTopics;
+import org.eclipse.hawkbit.ui.components.NotificationUnreadButton;
+import org.eclipse.hawkbit.ui.push.UIEventProvider;
+import org.springframework.util.CollectionUtils;
+import org.vaadin.spring.events.Event;
+import org.vaadin.spring.events.EventBus.UIEventBus;
+import org.vaadin.spring.events.EventBusListenerMethodFilter;
+import org.vaadin.spring.events.EventScope;
+import org.vaadin.spring.events.annotation.EventBusListenerMethod;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.vaadin.ui.UI;
+
+/**
+ * Remote events listener
+ */
+public class RemoteEventsListener {
```

Comment:
Wouldn't it better if this class is named `RemoteEventListener` and the internal `RemoteEventListener` get's a different name since it is only internal?

Reply:
Renamed `RemoteEventListener` to `HawkbitEntityEventListener`, internal `RemoteEventListener` clearly identifies the intention of the listener thus should not be renamed in my opinion